core/templates/dev/head/services/ExplorationHtmlFormatterService.js

```diff
@@ -33,7 +33,11 @@ oppia.factory('ExplorationHtmlFormatterService', [
           extensionTagAssemblerService.formatCustomizationArgAttrs(
             element, interactionCustomizationArgSpecs));
         element.attr('on-submit', 'submitAnswer(answer, rulesService);');
-        element.attr('last-answer', 'lastAnswer');
+        if (!canAcceptAnswers) {
+          element.attr('last-answer', 'null');
```

Comment:
Reduce the amount of stuff that's being branched on:

`element.attr('last-answer', canAcceptAnswers ? 'lastAnswer' : 'null');`

And add JSDoc + tests.

Reply:
Should I create a spec file for ExplorationHtmlFormatterService.js?

Reply:
Yes, that's a good idea.

Reply:
Done